### DISCUSSION TITLE
audio-4-channel-mic-96kHz

### DISCUSSION THREAD

**Discussion by MaximilianArlt** [hathach/tinyusb#2745] [Q&A]
### Operating System

Windows 11

### Board

STM32F7

### Firmware

examples/device/audio-4-channel-mic

### What happened ?

If I increase the sample rate to 96000 in the config.h, I don't get proper values, only something like noise. What could be the cause of this?

### How to reproduce ?

/

### Debug Log as txt file (LOG/CFG_TUSB_DEBUG=2)

/

### I have checked existing issues, dicussion and documentation

- [X] I confirm I have checked existing issues, dicussion and documentation.

**Comment by HiFiPhile** [comment]
Please do not use bug report for usage helping.

Also please provide more information.

**Reply by MaximilianArlt** [reply]
Okay, sry

I only used the 4-channel audio example and changed the sample rate from 48kHz to 96kHz in the config.h file; everything else remained the same as in the example. I only get noise. I even have noise now when I go back to 48kHz. In the config.h file, I also configured it as High-Speed, and nothing else.
![image](https://github.com/user-attachments/assets/bc90f8fc-34b0-4c8c-ac9e-24e2dd09d2ba)
![image](https://github.com/user-attachments/assets/a2a20320-86bb-4cd1-861c-efaf3850ec68)

**Reply by HiFiPhile** [reply]
Which board are you using ? Does it really support high-speed mode ?

**Reply by MaximilianArlt** [reply]
I use STM32F723e DISCOVERY Board and it support high-speed.

**Reply by HiFiPhile** [reply]
Something is wrong on your end, also tested with STM32723E-DISCO:

```diff
E:\mcu\tinyusb>git diff
diff --git a/examples/device/audio_4_channel_mic/src/plot_audio_samples.py b/examples/device/audio_4_channel_mic/src/plot_audio_samples.py
index d17a908b6..514a46777 100644
--- a/examples/device/audio_4_channel_mic/src/plot_audio_samples.py
+++ b/examples/device/audio_4_channel_mic/src/plot_audio_samples.py
@@ -9,7 +9,7 @@ if __name__ == '__main__':
     # differently from tested list below. Uncomment the next line to see full list and try to pick correct one
     # print(sd.query_devices())

-    fs = 48000                 # Sample rate
+    fs = 96000                 # Sample rate
     duration = 1   # Duration of recording

     if platform.system() == 'Windows':
diff --git a/examples/device/audio_4_channel_mic/src/tusb_config.h b/examples/device/audio_4_channel_mic/src/tusb_config.h
index 46484f847..c5d87521e 100644
--- a/examples/device/audio_4_channel_mic/src/tusb_config.h
+++ b/examples/device/audio_4_channel_mic/src/tusb_config.h
@@ -103,7 +103,7 @@ extern "C" {
 //--------------------------------------------------------------------

 // Have a look into audio_device.h for all configurations
-#define CFG_TUD_AUDIO_FUNC_1_SAMPLE_RATE              48000
+#define CFG_TUD_AUDIO_FUNC_1_SAMPLE_RATE              96000

 #define CFG_TUD_AUDIO_FUNC_1_DESC_LEN                 TUD_AUDIO_MIC_FOUR_CH_DESC_LEN

```
![image](https://github.com/user-attachments/assets/2731955c-2295-410d-a87a-ac49e0ab05a5)

**Reply by MaximilianArlt** [reply]
Okay, thank you. This is my "config.h" file, do you see anything wrong with that? 


#ifndef _TUSB_CONFIG_H_
#define _TUSB_CONFIG_H_


#ifdef __cplusplus
extern "C" {
#endif


//--------------------------------------------------------------------+
// Board Specific Configuration
//--------------------------------------------------------------------+
#define CFG_TUSB_MCU                 OPT_MCU_STM32F7
#define CFG_TUSB_OS                  1
#define BOARD_DEVICE_RHPORT_SPEED    OPT_MODE_HIGH_SPEED  // 480mbps
#define BOARD_DEVICE_RHPORT_NUM     1
#define CFG_TUSB_RHPORT1_MODE       (OPT_MODE_DEVICE | OPT_MODE_HIGH_SPEED)
#define BOARD_TUD_RHPORT			1
#define BOARD_TUD_MAX_SPEED			OPT_MODE_HIGH_SPEED
//#define TUP_USBIP_FSDEV

// RHPort number used for device can be defined by board.mk, default to port 0
#ifndef BOARD_TUD_RHPORT
#define BOARD_TUD_RHPORT      0
#endif

// RHPort max operational speed can defined by board.mk
#ifndef BOARD_TUD_MAX_SPEED
#define BOARD_TUD_MAX_SPEED   OPT_MODE_DEFAULT_SPEED
#endif

//--------------------------------------------------------------------
// COMMON CONFIGURATION
//--------------------------------------------------------------------

// defined by compiler flags for flexibility
#ifndef CFG_TUSB_MCU
#error CFG_TUSB_MCU must be defined
#endif

#ifndef CFG_TUSB_OS
#define CFG_TUSB_OS               OPT_OS_NONE
#endif

#ifndef CFG_TUSB_DEBUG
#define CFG_TUSB_DEBUG            0
#endif

// Enable Device stack
#define CFG_TUD_ENABLED           1

// Default is max speed that hardware controller could support with on-chip PHY
#define CFG_TUD_MAX_SPEED         BOARD_TUD_MAX_SPEED

/* USB DMA on some MCUs can only access a specific SRAM region with restriction on alignment.
 * Tinyusb use follows macros to declare transferring memory so that they can be put
 * into those specific section.
 * e.g
 * - CFG_TUSB_MEM SECTION : __attribute__ (( section(".usb_ram") ))
 * - CFG_TUSB_MEM_ALIGN   : __attribute__ ((aligned(4)))
 */
#ifndef CFG_TUSB_MEM_SECTION
#define CFG_TUSB_MEM_SECTION
#endif

#ifndef CFG_TUSB_MEM_ALIGN
#define CFG_TUSB_MEM_ALIGN        __attribute__ ((aligned(4)))
#endif

//--------------------------------------------------------------------
// DEVICE CONFIGURATION
//--------------------------------------------------------------------

#ifndef CFG_TUD_ENDPOINT0_SIZE
#define CFG_TUD_ENDPOINT0_SIZE    64
#endif

//------------- CLASS -------------//
#define CFG_TUD_AUDIO             1
#define CFG_TUD_CDC               0
#define CFG_TUD_MSC               0
#define CFG_TUD_HID               0
#define CFG_TUD_MIDI              0
#define CFG_TUD_VENDOR            0

//--------------------------------------------------------------------
// AUDIO CLASS DRIVER CONFIGURATION
//--------------------------------------------------------------------

// Have a look into audio_device.h for all configurations
#define CFG_TUD_AUDIO_ENABLE_INTERRUPT_EP			  1
#define CFG_TUD_AUDIO_FUNC_1_SAMPLE_RATE              96000

#define CFG_TUD_AUDIO_FUNC_1_DESC_LEN                 TUD_AUDIO_MIC_FOUR_CH_DESC_LEN

#define CFG_TUD_AUDIO_FUNC_1_N_AS_INT                 1
#define CFG_TUD_AUDIO_FUNC_1_CTRL_BUF_SZ              64

#define CFG_TUD_AUDIO_ENABLE_EP_IN                    1
#define CFG_TUD_AUDIO_FUNC_1_N_BYTES_PER_SAMPLE_TX    2         // This value is not required by the driver, it parses this information from the descriptor once the alternate interface is set by the host - we use it for the setup
#define CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX            4         // This value is not required by the driver, it parses this information from the descriptor once the alternate interface is set by the host - we use it for the setup
#define CFG_TUD_AUDIO_EP_SZ_IN                        TUD_AUDIO_EP_SIZE(CFG_TUD_AUDIO_FUNC_1_SAMPLE_RATE, CFG_TUD_AUDIO_FUNC_1_N_BYTES_PER_SAMPLE_TX, CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX)


#define CFG_TUD_AUDIO_ENABLE_ENCODING                 0
#define CFG_TUD_AUDIO_EP_IN_FLOW_CONTROL              1

#if CFG_TUD_AUDIO_ENABLE_ENCODING

#define CFG_TUD_AUDIO_FUNC_1_EP_IN_SZ_MAX             CFG_TUD_AUDIO_EP_SZ_IN
#define CFG_TUD_AUDIO_FUNC_1_EP_IN_SW_BUF_SZ          CFG_TUD_AUDIO_EP_SZ_IN

#define CFG_TUD_AUDIO_ENABLE_TYPE_I_ENCODING          1
#define CFG_TUD_AUDIO_FUNC_1_CHANNEL_PER_FIFO_TX      2         // One I2S stream contains two channels, each stream is saved within one support FIFO - this value is currently fixed, the driver does not support a changing value
#define CFG_TUD_AUDIO_FUNC_1_N_TX_SUPP_SW_FIFO        (CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX / CFG_TUD_AUDIO_FUNC_1_CHANNEL_PER_FIFO_TX)
#define CFG_TUD_AUDIO_FUNC_1_TX_SUPP_SW_FIFO_SZ       (TUD_OPT_HIGH_SPEED ? 64 : 8) * (CFG_TUD_AUDIO_EP_SZ_IN / CFG_TUD_AUDIO_FUNC_1_N_TX_SUPP_SW_FIFO) // Example write FIFO every 1ms, so it should be 8 times larger for HS device

#else

#define CFG_TUD_AUDIO_FUNC_1_EP_IN_SZ_MAX             CFG_TUD_AUDIO_EP_SZ_IN
#define CFG_TUD_AUDIO_FUNC_1_EP_IN_SW_BUF_SZ          32*CFG_TUD_AUDIO_EP_SZ_IN // Example write FIFO every 1ms, so it should be 8 times larger for HS device

#endif

#ifdef __cplusplus
}
#endif

#endif /* _TUSB_CONFIG_H_ */
